chromium_src/chrome/browser/component_updater/widevine_cdm_component_installer.cc

```diff
@@ -179,8 +188,38 @@ CrxInstaller::Result WidevineCdmComponentInstallerPolicy::OnCustomInstall(
                     "Consider removing our WIDEVINE_ARM64_DLL_FIX.";
     return CrxInstaller::Result(0);
   }
+  const std::string* version = manifest.FindString("version");
+  if (!version) {
+    return CrxInstaller::Result(update_client::InstallError::INVALID_VERSION);
+  }
+  // First try to guess a download URL that matches the component version. This
+  // usually works and ensures that the component and DLL versions match, even
+  // when upstream updates the component.
+  const GURL guessed_url = GURL(GetArm64DllZipUrl(*version));
+  CrxInstaller::Result result = InstallArm64Dll(guessed_url, install_dir);
+  if (IsHttp404(result)) {
+    // Our guess failed. Fall back to a DLL version that is known to exist.
+    const GURL fallback_url = GURL(kBraveWidevineArm64DllUrl.Get());
+    LOG(WARNING) << "Guessed Widevine Arm64 ALL URL " << guessed_url
+                 << " does not exist. Falling back to " << fallback_url
+                 << ", which should exist but may not be compatible.";
+    return InstallArm64Dll(fallback_url, install_dir);
+  }
+  return result;
+}
+
+std::string WidevineCdmComponentInstallerPolicy::GetArm64DllZipUrl(
```

Comment:
nit: this can be in anonymous namespace

Reply:
Resolved in 76c20793e3440af556b783a212c6c352f9b00fd0.